common/src/main/scala/org/apache/celeborn/common/metrics/MetricsSystem.scala

```diff
@@ -80,11 +80,14 @@ class MetricsSystem(
     MetricRegistry.name(source.sourceName)
   }
 
-  def getSourcesByName(sourceName: String): Seq[Source] =
+  def getSourcesByName(sourceName: String): Seq[Source] = sources.synchronized {
```

Comment:
In `registerSinks`, this `sources` gets passed to `PrometheusServlet` - and so could be used outside of the `synchronized` lock.

Would have been more ideal if this was a thread safe variant instead (like `CopyOnWriteArrayList`).
Is it possible to update the constructor signature of `prometheusServlet` ?

Either to `List<Source>` (`sourcesList` directly) or atleast to `Seq[Source]` (`sourcesList.asScala`)

Reply:
SGTM, will update the constructor signature of `PrometheusServlet`.

Reply:
addressed in https://github.com/apache/incubator-celeborn/pull/2023/commits/b5d7dff2e327757ca6bb386f950bf218f6692a32